src/main/java/com/microsoft/jenkins/appservice/commands/FileDeployCommand.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.jenkins.appservice.commands;
+
+import com.microsoft.azure.management.appservice.DeploymentSlot;
+import com.microsoft.azure.management.appservice.WebApp;
+import com.microsoft.jenkins.appservice.AzureAppServicePlugin;
+import com.microsoft.jenkins.appservice.util.Constants;
+import com.microsoft.jenkins.appservice.util.DeployTypeEnum;
+import com.microsoft.jenkins.azurecommons.command.CommandState;
+import com.microsoft.jenkins.azurecommons.command.IBaseCommandData;
+import com.microsoft.jenkins.azurecommons.command.ICommand;
+import com.microsoft.jenkins.azurecommons.telemetry.AppInsightsUtils;
+import hudson.FilePath;
+import hudson.Util;
+import org.apache.commons.lang.StringUtils;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FileDeployCommand implements ICommand<FileDeployCommand.IWarDeployCommandData> {
+    private String aiDeployInfo;
+    private String aiDeployFailedInfo;
+
+    @Override
+    public void execute(IWarDeployCommandData context) {
+        WebApp app = context.getWebApp();
+        DeployTypeEnum deployType = context.getDeployType();
+
+        FilePath srcDir = context.getJobContext().getWorkspace().child(Util.fixNull(context.getSourceDirectory()));
```

Comment:
null check on srcDir?

Reply:
child() method return a new FilePath, cannot be null.